### PR TITLE
container: Don't call runtime state on every container on list

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mpatel@redhat.com>

### CODE REVIEW

File: server/container_list.go [hunk 1]
```diff
@@ -66,10 +66,6 @@ func (s *Server) ListContainers(ctx context.Context, req *pb.ListContainersReque
 	}
 
 	for _, ctr := range ctrList {
-		if err := s.Runtime().UpdateStatus(ctr); err != nil {
-			return nil, err
-		}
-
 		podSandboxID := ctr.Sandbox()
 		cState := s.Runtime().ContainerStatus(ctr)
```
Comment (runcom) [inline server/container_list.go]: we still fork here right?

Comment (runcom) [inline server/container_list.go]: alright, reads in-memory

Comment (mrunalp) [inline server/container_list.go]: We just return in-memory state.

**Comment by rhatdan** [comment]
LGTM

**Comment by nalind** [comment]
The returned `ListContainerResponse`'s `Container` items have a `State` field in them, but if kubelet doesn't need that to be absolutely current every time (particularly if it calls `ContainerStatus()`, which also returns a `State`, and uses that), then there's no harm in not updating the state of every container in `ListContainers()`, so this'll be fine.

**Comment by rhatdan** [comment]
Is this pull request replaced by a newer one?

**Comment by rhatdan** [comment]
@mrunalp @runcom Should we close this?

**Comment by runcom** [comment]
It seems the inotify pr only addressed exited containers so this PR would still be needed. @mrunalp can correct me and close it otherwise

**Comment by rhatdan** [comment]
Ok, So the other patches fixed different problems. This could improve our performance more?

**Comment by runcom** [comment]
It should, I'm not sure this PR is ready though (or as said, maybe this is not needed)

**Comment by mrunalp** [comment]
This was covered in #766. Closing. We now have to address create spikes. steady state is much much better.

**Pull request closed without merging**